main/tests/server/src/com/google/refine/io/FileProjectManagerTests.java

```diff
@@ -92,4 +101,35 @@ public void saveReloadMultinationalCharacter() throws IOException {
         manager = new FileProjectManagerStub(workspaceDir);
         assertEquals(manager.getPreferenceStore().get("testPref"), "Refiné");
     }
+
+    /**
+     * Issue #1418 Issue #3719 Issue #3277 deleting the only existing project without saving the workspace will not
+     * remove the projectID from workspace.json
+     */
+    @Test
+    public void deleteProjectWithoutSavingWorkspace() throws IOException {
+        FileProjectManager manager = new FileProjectManagerStub(workspaceDir);
+        manager.saveToFile(workspaceFile);
+        manager.deleteProject(5555);
+
+        InputStream inputStream = new FileInputStream(workspaceFile);
+        JsonObject json = JSON.parse(inputStream);
+        assertFalse(json.get("projectIDs").getAsArray().isEmpty());
```

Comment:
I am not sure about this test: your comment above says "deleting the only existing project without saving the workspace will *not* remove the projectID from workspace.json" but in the test you do assert that the array is empty… That seems inconsistent, no? I think the test below should actually be sufficient, no?

Reply:
> but in the test you do assert that the array is empty… That seems inconsistent, no?

That is not the case. The test uses `assertFalse`  on `isEmpty()` which confirms that the array is not empty

> I think the test below should actually be sufficient, no?

Thats true since it confirms that deleting and saving the workspace removes projectID from wokspace.json thus fixing the issues

Reply:
Ah thanks, looks like I was not really awake :-D
Sounds good, then let's only keep the second test.